controller/gateway/controller_reconciler_utils.go

```diff
@@ -139,27 +137,39 @@ func (r *Reconciler) getGatewayAddresses(
 
 func gatewayConfigDataPlaneOptionsToDataPlaneOptions(
 	gatewayConfigNamespace string,
-	opts operatorv1beta1.GatewayConfigDataPlaneOptions,
+	opts GatewayConfigDataPlaneOptions,
 ) *operatorv1beta1.DataPlaneOptions {
 	dataPlaneOptions := &operatorv1beta1.DataPlaneOptions{
 		Deployment: opts.Deployment,
-		Extensions: opts.Extensions,
 	}
 
 	if len(opts.PluginsToInstall) > 0 {
 		dataPlaneOptions.PluginsToInstall = lo.Map(opts.PluginsToInstall,
-			func(pluginReference operatorv1beta1.NamespacedName, _ int) operatorv1beta1.NamespacedName {
+			func(pluginReference operatorv2alpha1.NamespacedName, _ int) operatorv1beta1.NamespacedName {
+				nn := operatorv1beta1.NamespacedName{
+					Name:      pluginReference.Name,
+					Namespace: pluginReference.Namespace,
+				}
+
 				// When Namespace is not provided, the GatewayConfiguration's namespace is assumed.
-				if pluginReference.Namespace == "" {
-					pluginReference.Namespace = gatewayConfigNamespace
+				if nn.Namespace == "" {
+					nn.Namespace = gatewayConfigNamespace
 				}
-				return pluginReference
+				return nn
 			},
 		)
 	}
 
 	if opts.Resources != nil {
-		dataPlaneOptions.Resources.PodDisruptionBudget = opts.Resources.PodDisruptionBudget
+		if opts.Resources.PodDisruptionBudget != nil {
```

Comment:
`if opts.Resources != nil && opts.Resources.PodDisruptionBudget != nil {` keeping code more to the left

Reply:
The way it's nested here we try to signal that there can be more `.Resources` fields that can be handled here.

Do you think it'd be better to change this given the above?